src/components/company/compare.tsx

```diff
@@ -10,28 +10,48 @@ import {
   TableRow,
 } from "@mui/material";
 import React from "react";
-import { ICompareResult } from "../../models/company/company.model";
+// import { ICompareResult } from "../../models/company/company.model";
+import { FinancialData } from "../../services/api";
+import { useOutletContext } from "react-router-dom";
+import { IAppContext } from "../../models";
 
 const Compare = () => {
-  const [compareResult, setCompareResult] = React.useState<ICompareResult[]>(
-    []
-  );
-  const [company1, setCompany1] = React.useState<string>("");
-  const [company2, setCompany2] = React.useState<string>("");
+  // const [compareResult, setCompareResult] = React.useState<ICompareResult[]>(
+  //   []
+  // );
+  const { api } = useOutletContext<IAppContext>();
+
+  const [company1FinancialData, setCompany1FinancialData] =
+    React.useState<FinancialData>();
+  const [company2FinancialData, setCompany2FinancialData] =
+    React.useState<FinancialData>();
+  const [company1, setCompany1] = React.useState<string>();
+  const [company2, setCompany2] = React.useState<string>();
+
+  const getCompany = React.useCallback(() => {
+    const loadCompanyData = async () => {
+      try {
+        if (!company1 || !company2) return;
 
-  const onCompare = React.useCallback(() => {
-    const dummyData: ICompareResult[] = [
-      {
-        total: 100,
-        name: company1,
-      },
-      {
-        total: 200,
-        name: company2,
-      },
-    ];
-    return setCompareResult(dummyData);
-  }, [company1, company2, setCompareResult]);
+        const response =
+          await api.enterprise.apiEnterpriseOndernemingsnummerFinancialYearGet(
```

Comment:
Dit is een concrete call. De intelisense stelt zelf voor welke calls beschikbaar zijn.